Restrictions/IntegerRestrictions.py

```diff
@@ -0,0 +1,136 @@
+        if minimum is None:
+            minimum = min_range_edge
+        elif minimum < min_range_edge:
+            raise Exception("minimum {} lower than the defined range's minimum {}".format(minimum, min_range_edge))
+
+        max_edge_range = getattr(self, 'MaxRangeEdge')
+        if maximum is None:
+            maximum = max_edge_range
+        elif maximum > max_edge_range:
+            raise Exception("maximum {} higher than the defined range's maximum {}".format(maximum, max_edge_range))
+
+        super().__init__(restriction_name_in_xml, minimum, maximum)
+
+
+class NonPositiveIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MaxRangeEdge = 0
+
+    def __init__(self, restriction_name_in_xml: str = "nonPositiveInteger", minimum: Optional[int] = None):
+        super().__init__(minimum=minimum, restriction_name_in_xml=restriction_name_in_xml)
+
+
+class NegativeIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MaxRangeEdge = -1
+
+    def __init__(self, restriction_name_in_xml: str = "negativeInteger", minimum: Optional[int] = None):
+        super().__init__(minimum=minimum, restriction_name_in_xml=restriction_name_in_xml)
+
+
+class NonNegativeIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = 0
+
+    def __init__(self, restriction_name_in_xml: str = "nonNegativeInteger", maximum: Optional[int] = None):
+        super().__init__(maximum=maximum, restriction_name_in_xml=restriction_name_in_xml)
+
+
+class PositiveIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = 1
+
+    def __init__(self, restriction_name_in_xml: str = "positiveInteger", maximum: Optional[int] = None):
+        super().__init__(maximum=maximum, restriction_name_in_xml=restriction_name_in_xml)
+
+
+class LongIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = -9223372036854775808
+    MaxRangeEdge = 9223372036854775807
+
+    def __init__(self, minimum: int = MinRangeEdge, maximum: int = MaxRangeEdge):
+        super().__init__("long", minimum, maximum)
+
+
+class IntIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = -2147483648
+    MaxRangeEdge = 2147483647
+
+    def __init__(self, minimum: int = MinRangeEdge, maximum: int = MaxRangeEdge):
+        super().__init__("int", minimum, maximum)
+
+
+class ShortIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = -32768
+    MaxRangeEdge = 32767
+
+    def __init__(self, minimum: int = MinRangeEdge, maximum: int = MaxRangeEdge):
+        super().__init__("short", minimum, maximum)
+
+
+class ByteIntegerRestriction(BaseInGivenRangeIntegerRestriction):
+    MinRangeEdge = -128
+    MaxRangeEdge = 127
+
+    def __init__(self, minimum: int = MinRangeEdge, maximum: int = MaxRangeEdge):
+        super().__init__("byte", minimum, maximum)
+
+
+class UnsignedLongIntegerRestriction(NonNegativeIntegerRestriction):
+    MaxRangeEdge = 18446744073709551615
```

Comment:
maybe add MinRangeEdge = 0?

Reply:
already done in the base class NonNegativeIntegerRestriction